packages/atomic/src/components/common/breadbox/breadbox.tsx

```diff
@@ -0,0 +1,241 @@
+import {FacetValueState} from '@coveo/headless';
+import {h, VNode, FunctionalComponent} from '@stencil/core';
+import CloseIcon from '../../../images/close.svg';
+import {FocusTargetController} from '../../../utils/accessibility-utils';
+import {Button} from '../button';
+import {AnyBindings} from '../interface/bindings';
+
+export interface IBreadcrumb {
+  facetId: string;
+  label: string;
+  formattedValue: string[];
+  state?: FacetValueState;
+  content?: VNode;
+  deselect: () => void;
+}
+
+type BreadboxFocusTargets = {
+  breadcrumbRemovedFocus: FocusTargetController;
+  breadcrumbShowLessFocus: FocusTargetController;
+  breadcrumbShowMoreFocus: FocusTargetController;
+};
+
+const SEPARATOR = ' / ';
+const ELLIPSIS = '...';
+
+function limitPath(path: string[], pathLimit: number) {
+  if (path.length <= pathLimit) {
+    return path.join(SEPARATOR);
+  }
+
+  if (pathLimit === 1 && path.length > 1) {
+    return [ELLIPSIS, path[path.length - 1]].join(SEPARATOR);
+  }
+
+  const ellipsedPath = [path[0], ELLIPSIS, ...path.slice(-(pathLimit - 1))];
+  return ellipsedPath.join(SEPARATOR);
+}
+
+export interface BreadcrumbButtonProps {
```

Comment:
This interface is very large: This points to a problem that this component (the button) is doing too much.

* `focusTargets` is un-necessary: what it can do instead is resurface the `ref` prop to the parent.
* Related to the above, `lastremoved`/`firstExpanded` then does not need to leak here, and can be handled by the top level web component.
* The content should be a separate component, and button could just output it's `children`.
* `facetId` does not seem to be used anywhere
* `deselect` does not seem to be used anywhere

Reply:
This comment potentially applies to all components in this file;